receiver/otelarrowreceiver/internal/metrics/otlp.go

```diff
@@ -37,7 +47,18 @@ func (r *Receiver) Export(ctx context.Context, req pmetricotlp.ExportRequest) (p
 	}
 
 	ctx = r.obsrecv.StartMetricsOp(ctx)
-	err := r.nextConsumer.ConsumeMetrics(ctx, md)
+
+	sizeBytes := int64(r.sizer.MetricsSize(req.Metrics()))
+	err := r.boundedQueue.Acquire(ctx, sizeBytes)
+	if err != nil {
+		return pmetricotlp.NewExportResponse(), err
+	}
+	defer func() {
+		releaseErr := r.boundedQueue.Release(sizeBytes)
+		r.logger.Error("Error releasing bytes from semaphore", zap.Error(releaseErr))
```

Comment:
Should this have `if releaseErr != nil`?

Reply:
Thanks for catching that!